packages/react-components/react-tree/src/components/TreeItem/TreeItem.types.ts

```diff
@@ -29,8 +31,12 @@ export type TreeItemProps = ComponentProps<Partial<TreeItemSlots>>;
 /**
  * State used in rendering TreeItem
  */
-export type TreeItemState = ComponentState<TreeItemSlots> &
-  TreeItemContextValue & {
-    open: boolean;
-    isLeaf: boolean;
-  };
+export type TreeItemState = ComponentState<TreeItemSlots> & {
+  open: boolean;
+  isLeaf: boolean;
+  /**
+   * By design, a button included on the actions slot should be small
+   */
+  buttonSize: 'small';
+  isActionsVisible: boolean;
```

Comment:
```suggestion
  areActionsVisible: boolean;
```

Reply:
I'm using the name `actions` as a **single** slot, although it's a plural name to indicate that multiple things could be added inside of it, it's technically a **single** slot